File: src/main/java/com/rdsms/service/OperatorService.java
package com.rdsms.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import com.rdsms.dao.IOperatorDAO;
import com.rdsms.entity.Operator;

/**
 * 
 * @author Nitesh
 *
 */

@Service
public class OperatorService implements IOperatorService{
	
	@Autowired
	private IOperatorDAO dao;

	@Override
	public List<Operator> getOperators() {
		return dao.getOperators();
	}

	@Override
	public Operator getOperatorById(int oepratorId) {
		return dao.getOperatorById(oepratorId);
	}

	@Override
	public Operator createOperator(Operator operator) {
		return dao.createOperator(operator);
	}

	@Override
	public Operator updateOperator(int operatorId, Operator operator) {
		return dao.updateOperator(operatorId, operator);
	}

	@Override
	public boolean deleteOperator(int operatorId) {
		return dao.deleteOperator(operatorId);
	}

}

File: rdsms-app/src/app/service/operator.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs/observable';
import {Http, Headers, RequestOptions, RequestMethod, Response } from '@angular/http';
import 'rxjs/add/operator/map';
import 'rxjs/add/operator/catch';
import { IOperator } from '../models/operator';
import { AppComponent } from '../app.component';
import { AuthService } from './auth.service';

/**
 * @author Nitesh
 */

@Injectable()
export class OperatorService {

  constructor(private http: Http, private service: AuthService) { }
  private endpoint = AppComponent.API_URL + '/operatorservice/operators';
  private options = this.service.getAuthHeaders();

  getOperators(): Observable<IOperator[]> {
    return this.http.get(this.endpoint, this.options)
          .map(this.extractData)
          .catch(this.handleError);
  }

  createOperator(operator) {
    console.log('Adding new operator into backend database..' + operator);
    return this.http.post(this.endpoint, operator, this.options)
    .map(success => success.status)
    .catch(this.handleError);
  }

  updateOperator(operator: IOperator): Observable<number> {
    console.log('updating operator in backend..' + operator);
    return this.http.put(this.endpoint + '/' + operator.operatorId, operator, this.options)
            .map(success => success.status)
            .catch(this.handleError);
  }

  deleteOperator(operatorId): Observable<number> {
    console.log('Deleting operator [id = ' + operatorId + '] from backend');
    return this.http.delete(this.endpoint + '/' + operatorId, this.options)
          .map(success => success.status)
          .catch(this.handleError);
  }

  private extractData(res: Response) {
    const body = res.json();
    return body;
  }

  private handleError (error: Response | any) {
    console.error(error.message || error);
    return Observable.throw(error.status);
  }

}
